docs/advanced_usage.rst

```diff
@@ -23,6 +23,88 @@ You can also use a Unix Domain Socket to connect to the agent::
     tracer.configure(uds_path="/path/to/socket")
 
 
+Context
+-------
+
+The :class:`Context` object is used to represent the state of a trace that can
+be used to propagate the trace across processes (:ref:`Distributed Tracing <disttracing>`).
+
+To retrieve the context of the currently active trace use::
+
+        context = tracer.current_trace_context()
+
+Note that if there is no active trace then ``None`` will be returned.
+
+
+Tracing Context Management
+--------------------------
+
+In ``ddtrace`` "context management" is the management of which :class:`Span` or
+:class:`Context` is active in an execution (thread, task, etc). There can only be one
+active span or context per execution.
+
+Context management enables parenting to be done implicitly when creating new
+spans by using the active span as the parent of a new span. When an active span
+finishes, its parent becomes the active span.
+
+``tracer.trace()`` automatically creates new spans as the child of the active
+context::
+
+    # Here no span is active
+    assert tracer.current_span() is None
+
+    with tracer.trace("parent") as parent:
+        # Here `parent` is active
+        assert tracer.current_span() is parent
+
+        with tracer.trace("child") as child:
+            # Here `child` is active.
+            # `child` automatically inherits from `parent`
+            assert tracer.current_span() is child
+
+        # `parent` is active again
+        assert tracer.current_span() is parent
+
+    # Here no span is active again
+    assert tracer.current_span() is None
+
+
+Manual Management
+^^^^^^^^^^^^^^^^^
+
+Parenting can be managed manually by using ``tracer.start_span()`` which by
+default does not activate spans when they are created::
+
+    with tracer.start_span("parent") as parent:
+        # Here no span is active
+        with tracer.start_span("parent", child_of=parent):
+            # Still no span active
+    # Still no span active
+
+    new_parent = tracer.start_span("new_parent", activate=True)
+    # new_parent is active
```

Comment:
we might want to make sure this is a fully valid use case. that is, creates 2 spans parent + child, starts, and finishes them both.

what do you think?

Reply:
Resolved offline: we should move these docs to tracer.start_span (since they're lacking there anyway) and then just point to the start_span docs here.

Reply:
Done in 679a128